### PR TITLE
Bringing APM branch closer to master

### PR DESCRIPTION
**Pull request by hiro2233**
Hi friends, this is the AntennaTracker v1,0,0 working on APM 2.x.

Dataflash is not tested, but i thinks it's will work very good. We need to test it,, if anyone could help testing it , it's will very fine for many people with this board. I have only generic boards meanwhile and i tested it very good.

Edit: misspelling.
Edit: Fixed APMRover and ArduPlane compilation.

Thanks in advance.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -89,4 +89,3 @@ serialsent.raw
 status.txt
 tags
 test.ArduCopter/*
-Thumbs.db
```
Comment (magicrub) [inline .gitignore]: why remove this? It should stay.

File: .travis.yml [hunk 2]
```diff
@@ -21,8 +21,7 @@ env:
     #   via the "travis encrypt" command using the project repo's public key
     - secure: "FjIwqZQV2FhNPWYITX5LZXTE38yYqBaQdbm3QmbEg/30wnPTm1ZOLIU7o/aSvX615ImR8kHoryvFPDQDWc6wWfqTEs3Ytq2kIvcIJS2Y5l/0PFfpWJoH5gRd6hDThnoi+1oVMLvj1+bhn4yFlCCQ2vT/jxoGfiQqqgvHtv4fLzI="
   matrix:
-    - TRAVIS_BUILD_TARGET="px4-v2"
-    - TRAVIS_BUILD_TARGET="sitl linux apm2 navio"
+    - TRAVIS_BUILD_TARGET="apm2"
```
Comment (magicrub) [inline .travis.yml]: A note to others: this looks totally wrong but keep in mind this is a PR for the master-AVR branch

File: AntennaTracker/version.h [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "ap_version.h"
+
+#define THISFIRMWARE "AntennaTracker V1.0.0"
+#define FIRMWARE_VERSION 1,0,0,FIRMWARE_VERSION_TYPE_BETA
```
Comment (magicrub) [inline AntennaTracker/version.h]: What does everyone else think about this?

**Comment by magicrub** [comment]
11 commits turns into +4776 / -502 lines. @hiro2233 the general rule is the larger the PR the longer it takes to get merged. I would strongly suggest you break this up into smaller "obviously correct" commits and/or PRs. You have a lot going on in this PR.

File: .gitignore (hunk 1 above)
Comment (hiro2233) [inline .gitignore]: I understand in the another PR that you comment to remove it. My mistakes, i misunderstood your comment on that PR. I will add it again.

File: AntennaTracker/version.h (hunk 3 above)
Comment (hiro2233) [inline AntennaTracker/version.h]: Ok, i thought the same, i will comment it.

**Comment by hiro2233** [comment]
This will end up being my nightmare :smile: 
Ok @magicrub i will do it. Thank you.

**Comment by rmackay9** [comment]
@hiro2233, you're saying that you're going to do all the support for AVR boards trying to run the latest Antenna Tracker, Plane, Rover, Copter?  Are you sure you're going to have time to do that?

I'm actually quite nervous about new official releases on the AVR boards and the backlash if they are of poor quality.  I actually don't think we should do "official" releases on these boards unless they've gone through the same rigourous beta testing that the final releases for the AVR boards went though.

**Comment by magicrub** [comment]
agreed @rmackay9

**Comment by gmorph** [comment]
I agree with Randy regarding the "backlash".  I will be porting some of the newer Rover features into the AVR branch as that's relatively straight forward for me and I can test appropriately.  However, I've no idea how your going to add features for Plane or Copter as there is basically no firmware space left on either for new features.

Thanks, Grant.

**Comment by magicrub** [comment]
I am in favor of slowly merging in feature/fixes from master into the avr branch.

**Comment by gmorph** [comment]
Again, the problem is new features simply will not fit into Plane or Copter.  You would have to remove some existing feature to get anything new to fit.  Bug fixes are fine so long as the code size isn't increased.  

Thanks, Grant.

**Comment by magicrub** [comment]
@gmorph Tracker/Rover would have more space than copter/plane

**Comment by gmorph** [comment]
Yep - Rover isn't a problem and Tracker even less of an issue and as I said I will be merging some of the recent changes into Rover.

**Comment by billbonney** [comment]
This seems completely wrong to me as it's using the 'git co master-AVR' to switch to a build variant (APM2). By modify all the build files to remove 'unwanted' platforms creates a major difference to the main repo, making it much harder to keep in step with those master branch developments.  'master-AVR' was essentially nothing more than a tag of the main repo. It allows for minor fixes to be introduced at the point in time where AVR builds stop being supported.

To maintain APM2 instep with current developments, you need checkout master and modify the build variant to build an APM2 binary. 

I think there are two options here. 1) fork the repo and head on down the path that these current changes are focused on (with a complex merge strategy between all the build and other files between to similar but not same code lines)
2) create the APM2 (AVR) build variant as based on the current code base. That variant build triggers the functional blocks required and creates an AVR binary

I guessing but 
Option 1) is whats is really happening i.e. Fork the repo (though we have this odd PR back to the main repo)
Option 2) Is actually what is desired and expected (but this option is much more complex to achieve)

My final opinion is that having a branch that is so different from master to be is essentially it's own master branch belongs in another repo.

**Comment by magicrub** [comment]
you make a good point. Something for the dev call!

**Comment by kwikius** [comment]
@hiro2233 
Great work  :) It is good to have some work being done on APM branch.
@billbonney 
In an ideal world there would be one branch, but it would require a drastically different approach.  I would base it on Generic Programming ( in a way similary to C++ STL) where algorithms are applied to Concepts not type) Perhaps then it is time to fork to try that out?  There is definitely a "High end " school ( most) and a "low end" school ( me :) ).  Maybe it is time for an ArdupilotLite or something like that which aims at the other end of the market. This would let the Linux guys get on with their versions more easily too.

**Comment by dgrat** [comment]
@gmorph 
I also think that it is important for users to be aware that the state of apm2 ist actually _not_ state of master.

**Comment by gmorph** [comment]
@grat - yes it might be worth adding a readme to the master-AVR branch and just explaining what the situation is.

**Comment by hiro2233** [comment]
> @hiro2233, you're saying that you're going to do all the support for AVR boards trying to run the latest Antenna Tracker, Plane, Rover, Copter? Are you sure you're going to have time to do that?
> 
> I'm actually quite nervous about new official releases on the AVR boards and the backlash if they are of poor quality. I actually don't think we should do "official" releases on these boards unless they've gone through the same rigourous beta testing that the final releases for the AVR boards went though.

I will try to run the latest Antenna Tracker and Rover. On Copter there is no space for APM 2.x to fit the code, Copter need 7% program space, and Plane has 3,5% free space left.

@rmackay9 No problem, i agree. i think we can test it rigorously and i'm in, i'm living with APM practically. All my applications are making and building with APM API, so i can contribute for the best ever and well.

It's not my first time working with a development team or a scene around the world, i know how is it. I was working with video game console modchips development and beta testing underground for making backups on XBOX 360, PlayStation and Nintendo hardwares, but i retired from the business.

The dev team have the last word, i'm doing the best for the team too. I like it.

**Comment by kwikius** [comment]
It is great to have APM branch being worked on . Keep up the good work :)

**Comment by hiro2233** [comment]
@billbonney 
It's the time, we need to integrate APM to expand in the robotics and automatization projects, for example, with APM API and URUS application, i'm developing a neighborhood safety and it's are syncronized with cloud system with each neighbour SamartPhone to alarm and tell who are needing a help in the neighborhood.

It's simple, people asking to me how we are going to do this and i answer, Ardupilot has the power.

**Comment by hiro2233** [comment]
I agree with @dgrat and @gmorph .

Thank you very much @kwikius .

**Comment by magicrub** [comment]
I changed the title of this PR because it had very little to do with Antenna Tracker, it touched many files significantly over all the projects.

@hiro2233 You know, I think I'm going to just say this. This PR will never be accepted. It changes too much. Please break it up and submit PRs that contain commits that change specific things. If that means many commits, that's fine. You can't just change everything all at once like this. It wouldn't hurt to re-read `this doc](http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html) too, specifically regarding what i just said above.

**Comment by hiro2233** [comment]
@magicrub on this oportunity i disagree with you Mr., because i'm working entirely with this new Antenna Tracker v1,0,0 version from master on APM 2.x hardware. And i'm adding some features to attach cloned servos on RC_Channel interactively from console regarding this issue https://github.com/ArduPilot/ardupilot/issues/4772 

Changing the title metadata are not searchable from search engine like google, please change the title again, i will try change this title again for well.

I readed many times online document development indeed.

Greetings.
Hiroshi.

**Pull request closed without merging**